baseband/vdif/tests/test_vdif.py

```diff
@@ -1468,14 +1467,24 @@ def test_via_stream_reader(self, tmpdir):
                 assert fw.tell() == fh.tell()
                 assert fw.time == fh.time
 
-            with vdif.open(out_file, 'rs', sample_rate=self.sample_rate) as fr:
-                assert fr.start_time == self.start_time
-                assert fr.sample_rate == self.sample_rate
-                assert fr.samples_per_frame == 1
-                assert fr.sample_shape == (2, 128)
-                data2 = fr.read()
-
-        assert_array_equal(data2, data1)
+        self.check_file(out_file)
+
+    def test_via_binary_modification(self, tmpdir):
+        out_file = str(tmpdir.join('upper128_binary_mod.vdif'))
+        # 1024 bytes payload + 32 bytes header = 1056 bytes = 264 words
+        binary = np.fromfile(SAMPLE_AROCHIME, 'u4').reshape(-1, 264)
```

Comment:
This is the fast change code. You can omit the construction of the header and just use the value below.

For 256 channels, you'd use `32+256`, `8<<24` here, and then do `:72` below.